### PR TITLE
Support training with batch normalization op; update training example

### PR DESCRIPTION
**Pull request by chudegao**
Update batch normalization  handler to support training 
Update training example:
  1. training tf model before save to to onnx model
  2. add vgg model to test larger mode performance
  3. remove optimizer op run when do test
  4. use all test data to run trained onnx model

### CODE REVIEW

File: onnx_tf/handlers/backend/batch_normalization.py [hunk 1]
```diff
@@ -27,33 +28,69 @@ def _common(cls, node, **kwargs):
...
-      return [cls.make_tensor_from_onnx_node(node, inputs=inputs)]
+    running_mean_1d = tensor_dict[node.inputs[3]]
+    running_var_1d = tensor_dict[node.inputs[4]]
+
+    # # from version 7, force to use test mode
+    # if cls.SINCE_VERSION >= 7 or node.attrs.get("is_test", 0):
+    #   inputs = [x, running_mean, running_variance, bias, scale]
+    #   return [cls.make_tensor_from_onnx_node(node, inputs=inputs)]
+
     spatial = node.attrs.get("spatial", 1) == 1
```
Comment (chinhuang007) [inline onnx_tf/handlers/backend/batch_normalization.py:49]: Please keep the old code for SINCE_VERSION < 7. No need to enable training there, but we should still support older versions.

Comment (chudegao) [reply onnx_tf/handlers/backend/batch_normalization.py:49]: The code commeted here is for SINCE_VERSION >= 7.  Do you mean we keep the code for  SINCE_VERSION < 7 like below?   Buts seems it's for training of SINCE_VERSION<7. 

if cls.SINCE_VERSION < 7 and not node.attrs.get("is_test", 0):
    spatial = node.attrs.get("spatial", 1) == 1
    momentum = node.attrs.get("momentum", 0.9)
    axis = [0] if spatial else [0] + list(range(2, total_num_dim))
    mean, variance = tf.nn.moments(x, axis)
    running_mean = running_mean * momentum + mean * (1 - momentum)
    running_variance = running_variance * momentum + variance * (1 - momentum)
    # TODO: need to conform to the documentation here
    inputs = [x, running_mean, running_variance, bias, scale]
    return [cls.make_tensor_from_onnx_node(node, inputs=inputs)]

Comment (chinhuang007) [reply onnx_tf/handlers/backend/batch_normalization.py:49]: Yes, some code is commented out, and some code is removed. The problem is the new code here doesn't work for the SINCE_VERSION < 7 models. Please check https://github.com/onnx/onnx/blob/master/docs/Changelog.md#BatchNormalization-6. As you can see, the node has an attribute "is_test" to indicate the use is for training or not. We should not change the behavior or the original intent with our own flag. I would suggest to leave the old code there as is and just handle training for SINCE_VERSION >= 7.

Comment (chudegao) [reply onnx_tf/handlers/backend/batch_normalization.py]: Updated, please help review.

**Review by chinhuang007 — APPROVED** [review]
LGTM!
